plugin/pkg/admission/podtolerationrestriction/admission.go

```diff
@@ -233,8 +233,13 @@ func (p *podTolerationsPlugin) getNamespace(nsName string) (*api.Namespace, erro
 			}
 			return nil, errors.NewInternalError(err)
 		}
+		if namespace.Name == "" {
```

Comment:
really not a fan of adding code here to workaround test failures

Reply:
i agree we should not add checks for some conditions in actual code path if those conditions are never supposed to occur. We should check why test code is not working as expected.

Reply:
wondering if the changes in https://github.com/kubernetes/kubernetes/pull/57504 to start making fake clients actually do watch-related things is causing problems

Reply:
wondering if the test flake started happening only after this PR.

Reply:
Adding this check without changing `admission_test.go` causes the test to fail:

```
$ ./podtolerationrestriction.test 
E0208 09:02:03.805252    7556 reflector.go:322] k8s.io/kubernetes/pkg/client/informers/informers_generated/internalversion/factory.go:86: Failed to watch *core.Namespace: unhandled watch: testing.WatchActionImpl{ActionImpl:testing.ActionImpl{Namespace:"", Verb:"watch", Resource:schema.GroupVersionResource{Group:"", Version:"", Resource:"namespaces"}, Subresource:""}, WatchRestrictions:testing.WatchRestrictions{Labels:labels.internalSelector(nil), Fields:fields.andTerm{}, ResourceVersion:""}}
--- FAIL: TestPodAdmission (0.10s)
        admission_test.go:262: Test: default cluster tolerations with empty pod tolerations and nil namespace tolerations, expected no error but got: got an empty namespace object from client
        admission_test.go:269: Test: default cluster tolerations with empty pod tolerations and nil namespace tolerations, expected: []core.Toleration{core.Toleration{Key:"testKey", Operator:"Equal", Value:"testValue", Effect:"NoSchedule", TolerationSeconds:(*int64)(nil)}} but got: []core.Toleration{}
        admission_test.go:275: Test: default cluster tolerations with empty pod tolerations and nil namespace tolerations, expected no error but got: got an empty namespace object from client
        admission_test.go:282: Test: default cluster tolerations with empty pod tolerations and nil namespace tolerations, expected: []core.Toleration{core.Toleration{Key:"testKey", Operator:"Equal", Value:"testValue", Effect:"NoSchedule", TolerationSeconds:(*int64)(nil)}} but got: []core.Toleration{}
FAIL
```

It appears that the way the test is written this object can sometimes be empty.

Fine with not adding checks to `admission.go`, but I think it demonstrates that the test is doing something weird, and we hit this when trying to add new test cases.